sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_client_connection.py

```diff
@@ -3325,3 +3383,8 @@ def _get_partition_key_definition(self, collection_link: str) -> Optional[Dict[s
             partition_key_definition = container.get("partitionKey")
             self.__container_properties_cache[collection_link] = _set_properties_cache(container)
         return partition_key_definition
+
+    def _add_request_context(self, request_context):
+        if http_constants.HttpHeaders.SessionToken in self.last_response_headers:
+            request_context['session_token'] = self.last_response_headers[http_constants.HttpHeaders.SessionToken]
+        self.last_response_headers["request_context"] = request_context
```

Comment:
We shouldn't be modifying this - this is not threadsafe; `self.last_response_headers` is pretty much deprecated, and very much should be treated as read-only and unsafe, because we never know what thread might be overwriting it, even as this method is acting on it.

Why are we adding input into the output response headers? How is this used?
We will need to find an alternative way to pass this information down the stack (if that is the intention).

Reply:
Removed request context as not needed